src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -514,6 +528,27 @@ internal NamedTypeSymbol UnderlyingTupleType
 
         internal bool HasFriendlyNames => _hasFriendlyNames;
 
+        public override ImmutableArray<ITypeSymbol> TupleElementTypes
+        {
+            get
+            {
+                return _fields.SelectAsArray(f => (ITypeSymbol)f.Type);
```

Comment:
On non-tuple types `TupleElementNames` will return an `ImmutableArray<string>` that has the value `default(ImmutableArray<string>)`.  This method returns `ImmutableArray<ITypeSymbol>.Empty`.  These are semantically different values (uninitialized array vs. empty array).  It seems incorrect to differ here.